src/train.py

```diff
@@ -100,113 +100,6 @@ def train_model(settings, hyp_params, train_loader, valid_loader, test_loader):
-
-            raw_loss = criterion(preds, eval_attr)
-
-            if reg_en:
-                # Sinkhorn Distances
-                # reg_term = torch.mul(h_list[-1], 0.1)
-                reg_term = h_list[-1]
-                combined_loss += hyp_params.reg_lambda * torch.mean(reg_term)
-            else:
-                combined_loss = raw_loss
-
-            combined_loss.backward()
-
-            torch.nn.utils.clip_grad_norm_(model.parameters(), hyp_params.clip)
-            optimizer.step()
-
-            proc_loss += raw_loss.item() * batch_size
-            proc_size += batch_size
-            epoch_loss += combined_loss.item() * batch_size
-            if i_batch % hyp_params.log_interval == 0 and i_batch > 0:
-                avg_loss = proc_loss / proc_size
-                elapsed_time = time.time() - start_time
-                print('Epoch {:2d} | Batch {:3d}/{:3d} | Time/Batch(ms) {:5.2f} | Train Loss {:5.4f}'.
-                      format(epoch, i_batch, num_batches, elapsed_time * 1000 / hyp_params.log_interval, avg_loss))
-                proc_loss, proc_size = 0, 0
-                start_time = time.time()
-
-            del text, audio, vision, eval_attr, preds, hiddens, h_list
-            gc.collect()
-            torch.cuda.empty_cache()
-
-        return epoch_loss / hyp_params.n_train
-
-    def evaluate(model, criterion, local_rank, test=False, train=False):
-        model.eval()
-        if train:
-            loader = train_loader
-        elif test:
-            loader = test_loader
-        else:
-            loader = valid_loader
-        total_loss = 0.0
-
-        results = []
-        truths = []
-        with torch.no_grad():
-            for i_batch, (batch_X, batch_Y, batch_META) in enumerate(loader):
-                sample_ind, text, audio, vision = batch_X
-                eval_attr = batch_Y.squeeze(dim=-1)  # if num of labels is 1
-
-                if hyp_params.use_cuda:
-                    if hyp_params.ddp:
-                        text, audio, vision, eval_attr = text.cuda(local_rank), audio.cuda(local_rank), vision.cuda(
-                            local_rank), eval_attr.cuda(local_rank)
-                    else:
-                        text, audio, vision, eval_attr = text.cuda(), audio.cuda(), vision.cuda(), eval_attr.cuda()
-
-                batch_size = text.size(0)
-                preds, last_h_proj, h_list = model(text, audio, vision)
-                total_loss += criterion(preds, eval_attr).item() * batch_size
-
-                # add reg on validation set to check loss change
-                if not test and (hyp_params.reg_en is True):
-                    reg_term = h_list[-1]
-                    total_loss += hyp_params.reg_lambda * torch.mean(reg_term)
-                # Collect the results into dictionary
-                results.append(preds)
-                truths.append(eval_attr)
-
-        avg_loss = total_loss / (hyp_params.n_test if test else hyp_params.n_valid)
```

Comment:
It seems a bug her. The data loader has 3 options: train, valid, test. But here the denominator only has 2.